src/client/api/v1/get_balance.c

```diff
@@ -0,0 +1,172 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <string.h>
+
+#include "client/api/json_utils.h"
+#include "client/api/v1/get_balance.h"
+#include "client/api/v1/response_error.h"
+#include "client/network/http.h"
+#include "core/utils/iota_str.h"
+
+int deser_balance_info(char const *const j_str, res_balance_t *res) {
+  char const *const key_addr = "address";
+  char const *const key_maxResults = "maxResults";
+  char const *const key_count = "count";
+  char const *const key_balance = "balance";
+  char const *const key_code = "code";
+  char const *const key_message = "message";
+  double *number;
+  int ret = 0;
+
+  cJSON *json_obj = cJSON_Parse(j_str);
+  if (json_obj == NULL) {
+    return -1;
+  }
+
+  cJSON *error_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_error);
+  if (error_obj) {
+    char *code = calloc(32, sizeof(char));
+
+    // gets err code
+    if ((ret = json_get_string(error_obj, key_code, code, 32)) != 0) {
+      printf("[%s:%d]: gets %s json error code failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+    }
+
+    if (strcmp(code, "invalid_data") == 0) {
+      printf("[%s:%d]: http code 400\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+
+    } else if (strcmp(code, "not_found") == 0) {
+      printf("[%s:%d]: http code 404\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+    }
+
+    free(code);
+  }
+
+  cJSON *data_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_data);
+  if (data_obj) {
+    // gets addr
+    char *addr = calloc(64, sizeof(char));
+    if ((ret = json_get_string(data_obj, key_addr, addr, 64)) != 0) {
+      printf("[%s:%d]: gets %s json addr failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    if (strlen(addr) == 64) {
+      byte_t addr_bytes[IOTA_ADDRESS_BYTES];
+      hex2bin(addr, addr_bytes + 1, IOTA_ADDRESS_BYTES - 1);
+      addr_bytes[0] = 0x0001;  // Ed25519
+      memcpy(res->addr, addr_bytes, IOTA_ADDRESS_BYTES);
```

Comment:
I think we can do it in-place.
```suggestion
      hex2bin(addr, res->addr + 1, IOTA_ADDRESS_BYTES - 1);
      res->addr[0] = ADDRESS_VER_ED25519;  // Ed25519
```
or keeps the address as a string in this response object.